File: FliprrInternProject/frontend/src/services/newsletterService.js
import api from '../config/api'

export const newsletterService = {
  subscribe: async (email) => {
    const response = await api.post('/newsletter/subscribe', { email })
    return response.data
  },

  getAllSubscribers: async () => {
    const response = await api.get('/newsletter/subscribers')
    return response.data
  },

  deleteSubscriber: async (id) => {
    const response = await api.delete(`/newsletter/subscribers/${id}`)
    return response.data
  }
}

File: FliprrInternProject/backend/src/main/java/com/nextinvest/service/NewsletterService.java
package com.nextinvest.service;

import com.nextinvest.entity.Newsletter;
import com.nextinvest.repository.NewsletterRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class NewsletterService {
    
    @Autowired
    private NewsletterRepository newsletterRepository;
    
    public Newsletter subscribe(String email) {
        if (newsletterRepository.existsByEmail(email)) {
            throw new RuntimeException("Email already subscribed");
        }
        
        Newsletter newsletter = new Newsletter();
        newsletter.setEmail(email);
        return newsletterRepository.save(newsletter);
    }
    
    public List<Newsletter> getAllSubscribers() {
        return newsletterRepository.findAll();
    }
    
    public void deleteSubscriber(Long id) {
        newsletterRepository.deleteById(id);
    }
}
